dataplane/dregress.h

```diff
@@ -43,44 +43,76 @@ using direction_t = std::tuple<common::ip_prefix_t, ///< prefix
                                uint32_t, ///< peer_as
                                uint32_t>; ///< origin_as
 
-}
+using ConnTable = dataplane::hashtable_chain_spinlock_t<dregress::connection_key_t,
+                                                        dregress::connection_value_t,
+                                                        YANET_CONFIG_DREGRESS_HT_SIZE,
+                                                        YANET_CONFIG_DREGRESS_HT_EXTENDED_SIZE,
+                                                        4,
+                                                        4>;
+
+struct LimitsStats
+{
+	uint64_t pairs;
+	uint64_t keysSize;
+	uint64_t extendedChunksCount;
+	LimitsStats& operator+=(const LimitsStats& other)
+	{
+		pairs += other.pairs;
+		keysSize += other.keysSize;
+		extendedChunksCount += other.extendedChunksCount;
+		return *this;
+	}
+};
+
+} // namespace dregress
+
+namespace dataplane
+{
+class SlowWorker;
+} // namespace dataplane
 
 class dregress_t
 {
 public:
-	dregress_t(cControlPlane* controlplane, cDataPlane* dataplane);
+	dregress_t(dataplane::SlowWorker* slow, cDataPlane* dataplane, uint32_t gc_step);
+	dregress_t(dregress_t&& other);
 	~dregress_t();
 
+	dregress_t& operator=(dregress_t&& other);
+
 	void insert(rte_mbuf* mbuf);
 	void handle();
 
 	std::optional<dregress::direction_t> lookup(rte_mbuf* mbuf);
 	bool tcp_parse(rte_mbuf* mbuf, uint16_t& rtt, uint32_t& loss_count, uint32_t& ack_count);
-
-	common::idp::get_dregress_counters::response get_dregress_counters();
-	void limits(common::idp::limits::response& response);
+	dregress::LimitsStats limits() const;
+	const common::dregress::stats_t& Stats() const { return stats; }
+	const dregress::ConnTable* Connections() const { return connections; }
+	[[nodiscard]] std::lock_guard<std::mutex> LockCounters() { return std::lock_guard{counters_mutex}; }
+	const common::dregress::counters_t& Counters4() { return counters_v4; }
+	const common::dregress::counters_t& Counters6() { return counters_v4; }
```

Comment:
`const`

Reply:
and i would have thought the main problem is that both return v4 counters...